applications/skyekiwi-protocol.md

```diff
@@ -0,0 +1,200 @@
+* **Total Costs:** $30,000 (payable in DAI)
+
+### Milestone 1 - Phase 1
+
+**Phase One** - PoC Phase: Connects to a smart contract of a Substrate style WASM smart contract enabled blockchain; the Crust Network for storage; Public IPFS Pin Service & Gateways, with a SkyeKiwi ran IPFS Cluster as backup IPFS nodes. 
+
+* **Estimated duration:** 2 month
+* **FTE:**  2
+* **Costs:** 12,000USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can integrate the SkyeKiwi Protocol into their project. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests on both Node.js and Browser environment. |
+| 1. | Core Protocol | Implements the `upstream` `downstream` `updateEncryptionSchema` and `verifyProofOfAccess` APIs. |
+| 2. | Smart Contract | Implements and test for the smart contract used for registering secrets and access control write access to the secret. |
+| 3. | Polkadot.js | Add in encryption/decryption functionality to `@polkadot/keyring` and `@polkadot/extension` so that the protocol can run without the needs to read the private key of users. |
+### Milestone 2 - Phase 2
+
+**Phase Two**: Connects to the SkyeKiwi Network `pallet-secrets` instead of a smart contract for general secrets registry; the Crust Network for storage; Public IPFS Pin Service & temporary IPFS nodes from the Crust Network ecosystem solutions. 
+
+**Note:** We won't be 100% sure of the availability of the Crust Network temporary IPFS solution but will actively engage with the Crust Network for this part. Therefore, we'd like to mark it as optional. Worst thing worst, we still have our own IPFS cluster running. 
+
+* **Estimated duration:** 2 month
+* **FTE:**  2
+* **Costs:** 18,000USD
+
+| Number | Deliverable         | Specification                                                |
+| -----: | ------------------- | ------------------------------------------------------------ |
+|    0a. | License             | Apache 2.0                                                   |
+|    0b. | Documentation       | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can integrate the SkyeKiwi Protocol into their project. |
+|    0c. | Testing Guide       | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests on both Node.js, Browser and Rust environment. |
+|     1. | `pallet-secrets`    | Port the `ink!` smart contract to be a runtime pallet and integrate it to the SkyeKiwi Protocol source code. |
+|     2. | Rust Implementation | Port the SkyeKiwi Protocol to Rust, so that it works between blockchain nodes as well. |
```

Comment:
This would be deliverable 1 in M1? What do you mean by "between blockchain nodes"?

Reply:
No. M1 is implementing a client side version to run in browsers. M2-2 is a Rust implementation to run the protocol in Substrate. "Between blockchain nodes": encrypted by the identity key of nodes and able to share encrypted message between nodes with little coordination. However, the Rust version will be a straight & plain port, the way to encrypt with identity key and coordination will not be covered in this application.